Skip "test-rwlock1" when multithreading is disabled, which is the case
during bootstrapping on architectures not supported by GNU Mes.

Taken from upstream gnulib:
https://git.savannah.gnu.org/cgit/gnulib.git/commit/?id=fdff8bd09a7f053381f8bdb107ab5280b7c95959

diff --git a/gnulib-tests/test-rwlock1.c b/gnulib-tests/test-rwlock1.c
--- a/gnulib-tests/test-rwlock1.c
+++ b/gnulib-tests/test-rwlock1.c
@@ -21,6 +21,8 @@
 
 #include <config.h>
 
+#if USE_ISOC_THREADS || USE_POSIX_THREADS || USE_ISOC_AND_POSIX_THREADS || USE_WINDOWS_THREADS
+
 #include "glthread/lock.h"
 
 #include <errno.h>
@@ -151,3 +153,18 @@ main ()
       sleep (1);
     }
 }
+
+#else
+
+/* No multithreading available.  */
+
+#include <stdio.h>
+
+int
+main ()
+{
+  fputs ("Skipping test: multithreading not enabled\n", stderr);
+  return 77;
+}
+
+#endif

